### PR TITLE
Add notes about running the UI test suite

### PR DESCRIPTION
**Pull request by bstrie**
As requested by @jyn514 .

### CODE REVIEW

File: src/tests/running.md [hunk 1]
```diff
@@ -59,6 +59,43 @@ Likewise, you can test a single file by passing its path:
...
+If you have made a change to the way that the compiler formats its output,
+it would be extraordinarily tedious to have to reformat every `*.stderr`
+file manually. Fortunately there exists a flag that will allow you to
+"bless" the current output of the compiler as the expected output.
+When using this flag with the test runner, any UI tests that would
+ordinarily fail will instead have their `*.stderr` files overwritten
+with whatever output the compiler produces:
+
+```bash
+./x.py test src/test/ui --bless
```
Comment (jyn514) [inline src/tests/running.md:84]: Most of this info is already present in the '## Editing and updating the reference files' section below. Can you move it to that section so it's not duplicated?

File: src/tests/running.md [hunk 2]
```diff
@@ -59,6 +59,43 @@ Likewise, you can test a single file by passing its path:
...
+```
+
+Some UI tests will have different output depending on which "mode" that
+the compiler is in. Specifically, the compiler may have different output
+depending on whether the feature "non-lexical lifetimes" (NLL) is enabled.
+Any UI test such as `foo.rs` whose output differs with NLL enabled will
+have both a `foo.stderr` file and a `foo.nll.stderr` file.
+By default, the UI test suite will not be run in NLL mode.
```
Comment (jyn514) [inline src/tests/running.md:92]: This makes it sound like `nll` is the only mode, but there are other modes like `noopt` and `migrate`; plus any revisions will also have their own stderr files. Could you change it to use nll as an example instead?
```suggestion
the compiler is in. For example, "non-lexical lifetimes" (NLL) mode or a [revision](./adding.html#revisions) will have a different output file than the default.
```

**Comment by bstrie** [comment]
Ah yes, I forgot that the document already talked about --bless. I didn't see anything that my section would add over the original so I just removed it. I've moved the compare-modes bit into its own new section at the bottom and reworded it.

**Review by jyn514 — APPROVED** [review]
Looks great, thanks!

**Pull request merged**